### ISSUE TITLE
Integrate progress bar and pipeline and parallelize reads

### ISSUE THREAD

**Issue by kerneltime** [oneconcern/datamon#94]
Reading large files can lead to a poor user experience.
1. Integrate with https://github.com/cheggaaa/pb to give better feedback
2. look into parallelize CAFS reads and use of io.pipes

**Comment by chandresh-pancholi** [comment]
Is this for bundle upload and download?

**Comment by ransomw1c** [comment]
it'd be good to have a proposal for user feedback for large files.  cafs parallelization is already done.

**Comment by kerneltime** [comment]
https://github.com/oneconcern/datamon/pull/99 and other PRs cover parallelize reads.
Without having a clear ask for how to better represent progress, the current CLI works file as the blobs are uploaded and logged in parallel.

**Comment by ransomw1c** [comment]
there's been an ask from Zhuo for a clearer way to indicate progress on uploads
